src/main/java/org/opensearch/security/privileges/PermissionBasedPrivilegesEvaluationContext.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.privileges;
+
+import java.util.function.Supplier;
+
+import com.google.common.collect.ImmutableSet;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.security.action.apitokens.Permissions;
+import org.opensearch.security.resolver.IndexResolverReplacer;
+import org.opensearch.security.user.User;
+import org.opensearch.tasks.Task;
+
+public class PermissionBasedPrivilegesEvaluationContext extends PrivilegesEvaluationContext {
+    private final Permissions permissions;
+
+    public PermissionBasedPrivilegesEvaluationContext(
+        User user,
+        String action,
+        ActionRequest request,
+        Task task,
+        IndexResolverReplacer indexResolverReplacer,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<ClusterState> clusterStateSupplier,
+        Permissions permissions
+    ) {
+        super(user, action, request, task, indexResolverReplacer, indexNameExpressionResolver, clusterStateSupplier);
+        this.permissions = permissions;
+    }
+
+    @Override
+    public String toString() {
+        return "PermissionBasedPrivilegesEvaluationContext{"
+            + "user="
+            + getUser()
+            + ", action='"
+            + getAction()
+            + '\''
+            + ", request="
+            + getRequest()
+            + ", resolvedRequest="
+            + getResolvedRequest()
+            + ", permissions="
+            + permissions
+            + '}';
+    }
+
+    public Permissions getPermissions() {
+        return permissions;
+    }
+
+    @Override
+    public ImmutableSet<String> getMappedRoles() {
+        return ImmutableSet.of();
```

Comment:
There are quite a few dependencies on mapped roles.

- There's `ProtectedIndexEvaluator` which uses the setting `plugins.security.protected_indices.roles` to define roles which can access protected indices
- `PrivilegeEvaluator` uses mapped roles to realize the Dashboards multi tenancy implementation
- User attributes exposes the mapped roles via `${user.securityRoles}`, thus they can be used in DLS statements

Is it anticipated that this functionality won't be available with API tokens?